File: contest/ICPC/2018/北美洲/北美洲中北部/data/eulersnumber/submissions/accepted/e.py
#!/usr/bin/env python3

def euler(n):
    # mistake file here for range(n) not n+1
    val = 0
    # mistake: make sure someone isn't calling factorial each time
    fact = 1
    for i in range(n+1):
        if i != 0:
            fact *= i
        val += 1/fact
    return val

n = int(input())
print(euler(n))

File: contest/ICPC/2018/北美洲/北美洲中北部/data/eulersnumber/submissions/accepted/E.java

import java.util.Scanner;

public class E {

    public static void main(String[] args) {
	Scanner in = new Scanner(System.in);
	int n;
	n = in.nextInt();
	
	System.out.println(euler(n));
    }
    
    static double euler(int n) {
	double val = 0;
	
	long fact = 1;

	if(17 <= n+1)
	    n = 17;
	
	for(int i=0; i<n+1; i++) {
	    if(i != 0)
		fact *= i;
	    val += 1.0/fact;
	}
	
	return val;
    }
}
